### PR TITLE
ThingSpeak Adaptor

### PR DESCRIPTION
**Pull request by giovannipollo**

### CODE REVIEW

File: src/services/thingspeakadaptor/app.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        if len(path) == 0:
+            return self.asjson_info("ThingSpeak Adaptor Endpoint")
+        elif path[0] == "temperaturewrite":
+            r = requests.get(f"https://api.thingspeak.com/update?api_key={self._thingspeakapikeytemperaturewrite}&field1={args['temp']}")
+            if r.status_code != 200:
+                cherrypy.response.status = 400
+                return self.asjson_error({"response": r.json()})
+            return self.asjson_info(None)
+        elif path[0] == "humiditywrite":
```
Comment (thegabriele97) [inline src/services/thingspeakadaptor/app.py]: Wouldn't be better to use a POST method on a single endpoint? 

For example

GET /humidity: returns humidity as `humidityread` does actually
POST /humidity: what `humiditywrite` does

Comment (giovannipollo) [reply src/services/thingspeakadaptor/app.py]: I like the idea you suggest. However, when I created the channel, I found this instruction, as you can see in the image:

Screenshot 2022-03-05 at 16 54 22

Unluckily, as you can see, there are different instructions for reading and writing a channel. I cover the image with black square just to preserve the API key. 

If you have any other idea to simplify the code (I didn't find any) and use a single endpoint with different requests, let me know!

Comment (thegabriele97) [reply src/services/thingspeakadaptor/app.py]: Ok but this is not a problem for us. I mean, you can do something like this (untested):
```python
 @cherrypy.tools.json_out()
    def GET(self, *path, **args):

        if len(path) == 0:
            return self.asjson_info("ThingSpeak Adaptor Endpoint")
        elif path[0] == "temperature":
            r = requests.get(f"https://api.thingspeak.com/channels/{self._channelidtemperature}/feeds.json?api_key={self._thingspeakapikeytemperatureread}&results={args['hum']}")
            if r.status_code != 200:
                cherrypy.response.status = 400
                return self.asjson_error({"response": r.json()})
            return r.json()['feeds']
        elif path[0] == "humidity:
            r = requests.get(f"https://api.thingspeak.com/channels/{self._channelidhumidity}/feeds.json?api_key={self._thingspeakapikeyhumidityread}&results={args['hum']}")
            if r.status_code != 200:
                cherrypy.response.status = 400
                return self.asjson_error({"response": r.json()})
            return r.json()['feeds']

 @cherrypy.tools.json_out()
    def POST(self, *path, **args):

        if len(path) == 0:
            return self.asjson_info("ThingSpeak Adaptor Endpoint")
        elif path[0] == "temperature":
            r = requests.get(f"https://api.thingspeak.com/update?api_key={self._thingspeakapikeytemperaturewrite}&field1={args['temp']}")
            if r.status_code != 200:
                cherrypy.response.status = 400
                return self.asjson_error({"response": r.json()})
            return self.asjson_info(None)
        elif path[0] == "humidity":
            r = requests.get(f"https://api.thingspeak.com/update?api_key={self._thingspeakapikeyhumiditywrite}&field1={args['hum']}")
            if r.status_code != 200:
                cherrypy.response.status = 400
                return self.asjson_error({"response": r.json()})
            return self.asjson_info(None)
```

Comment (giovannipollo) [reply src/services/thingspeakadaptor/app.py]: Yes, you are right. Tomorrow I will implement it and test it. Thanks for the suggestions

**Review by eueu1997 — APPROVED** [review]

**Review by francescasilvano — APPROVED** [review]

File: src/services/thingspeakadaptor/app.py (hunk 1 above)
Comment (giovannipollo) [reply src/services/thingspeakadaptor/app.py]: I made it and it works properly. Thanks for the suggestion. I will resolve the conversation

**Comment by thegabriele97** [comment]
@giop98 do we want to wait for you to implement the MQTT funcionalities before merging this PR?

**Review by thegabriele97 — APPROVED** [review]

**Pull request merged**